appIconIndicators.js

```diff
@@ -941,14 +944,17 @@ var DominantColorExtractor = class DashToDock_DominantColorExtractor {
         if (iconTexture instanceof Gio.FileIcon) {
             // Use GdkPixBuf to load the pixel buffer from the provided file path
             return GdkPixbuf.Pixbuf.new_from_file(iconTexture.get_file().get_path());
+        } else if (iconTexture instanceof Gio.ThemedIcon) {
+            // Get the pixel buffer from the icon theme
+            let icon_info = themeLoader.lookup_icon(iconTexture.get_names()[0], DOMINANT_COLOR_ICON_SIZE, 0);
```

Comment:
Please handle the case we can't find any icon.

Reply:
The case where you can't find any icon within the `Gio.ThemedIcon`?

Reply:
I mean if `icon_info` is null (also use `const` for it).

Reply:
Got it, do you think it would be better to use then [`choose_icon`](https://lazka.github.io/pgi-docs/Gtk-3.0/classes/IconTheme.html#Gtk.IconTheme.choose_icon) instead of [`lookup_icon`](https://lazka.github.io/pgi-docs/Gtk-3.0/classes/IconTheme.html#Gtk.IconTheme.lookup_icon) or try to iterate over the names ourselves?